pkg/typesystem/typesystem.go

```diff
@@ -252,25 +252,25 @@ func (t *TypeSystem) validateRelationTypeRestrictions() error {
 	return nil
 }
 
-func isAssignable(rewrite *openfgapb.Userset) bool {
+func IsAssignable(rewrite *openfgapb.Userset) bool {
```

Comment:
Can we please refactor this so that it's not associated with a rewrite but instead a relation itself.

e.g.

```
func (t *TypeSystem) IsDirectlyAssignable(relation *openfgapb.Relation) bool {
    rewrite := relation.GetUsersetRewrite()

    return containsSelf(rewrite)
}

func containsSelf(rewrite *openfgapb.Userset) bool {
    // copy/paste the current logic here
}
```
It's weird to me that we have an IsAssignable method that applies to a rewrite but not a Relation. Rewrites aren't assignable or not, relations are.

Reply:
I can do that, but then I need to rollback my use of `typesystem.IsAssignable` in Write like you suggested, because I don't have the relation readily available there.